nifi-toolkit/nifi-toolkit-zookeeper-migrator/src/main/java/org/apache/nifi/toolkit/zkmigrator/ZooKeeperMigrator.java

```diff
@@ -269,15 +269,23 @@ private String ensureNodeExists(ZooKeeper zooKeeper, String path, CreateMode cre
         }
     }
 
-    private DataStatAclNode transformNode(DataStatAclNode node, AuthMode destinationAuthMode) {
-        // For the NiFi use case, all nodes will be migrated to CREATOR_ALL_ACL
+    private DataStatAclNode transformNode(DataStatAclNode node, AuthMode destinationAuthMode, boolean useExistingACL) {
+        // If useExistingACL is false, for the NiFi use cases, all nodes will be migrated to CREATOR_ALL_ACL
         final DataStatAclNode migratedNode = new DataStatAclNode(node.getPath(), node.getData(), node.getStat(),
-                destinationAuthMode.equals(AuthMode.OPEN) ? ZooDefs.Ids.OPEN_ACL_UNSAFE : ZooDefs.Ids.CREATOR_ALL_ACL,
+                determineNodeACL(node,destinationAuthMode,useExistingACL),
                 node.getEphemeralOwner());
         LOGGER.info("transformed original node {} to {}", node, migratedNode);
         return migratedNode;
     }
 
+    private List<ACL> determineNodeACL(DataStatAclNode node, AuthMode destinationAuthMode, boolean useExistingACL){
```

Comment:
@jtstorck Rather than have a separate stage I ended up combining this with the previous method.  Didn't really see the value of me having it separated out in the first place .